flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -342,6 +342,15 @@ public void alterCatalog(String catalogName, CatalogDescriptor catalogDescriptor
         if (catalogStore.contains(catalogName) && oldCatalogDescriptor.isPresent()) {
             Configuration conf = oldCatalogDescriptor.get().getConfiguration();
             conf.addAll(catalogDescriptor.getConfiguration());
+            catalogDescriptor
```

Comment:
It'd be better to make the configuration change as an updater rather than directly passing the catalog descriptor here.

E.g.

```java
    /**
     * Alters a catalog under the given name. The catalog name must be unique.
     *
     * @param catalogName the given catalog name under which to alter the given catalog
     * @param catalogUpdater catalog configuration updater to alter catalog
     * @throws CatalogException If the catalog neither exists in the catalog store nor in the
     *     initialized catalogs, or if an error occurs while creating the catalog or storing the
     *     {@link CatalogDescriptor}
     */
    public void alterCatalog(String catalogName, Consumer<Configuration> catalogUpdater)
            throws CatalogException {
        checkArgument(
                !StringUtils.isNullOrWhitespaceOnly(catalogName),
                "Catalog name cannot be null or empty.");
        checkNotNull(catalogUpdater, "Catalog configuration updater cannot be null.");

        CatalogStore catalogStore = catalogStoreHolder.catalogStore();
        Optional<CatalogDescriptor> oldCatalogDescriptor = getCatalogDescriptor(catalogName);

        if (catalogStore.contains(catalogName) && oldCatalogDescriptor.isPresent()) {
            Configuration conf = oldCatalogDescriptor.get().getConfiguration();
            catalogUpdater.accept(conf);
            CatalogDescriptor newCatalogDescriptor = CatalogDescriptor.of(catalogName, conf);
            Catalog newCatalog = initCatalog(catalogName, newCatalogDescriptor);
            catalogStore.removeCatalog(catalogName, false);
            if (catalogs.containsKey(catalogName)) {
                catalogs.get(catalogName).close();
            }
            newCatalog.open();
            catalogs.put(catalogName, newCatalog);
            catalogStoreHolder.catalogStore().storeCatalog(catalogName, newCatalogDescriptor);
        } else {
            throw new CatalogException(
                    String.format("Catalog %s does not exist in the catalog store.", catalogName));
        }
    }
```

For `ALTER CATALOG SET`
```java
    @Override
    public TableResultInternal execute(Context ctx) {
        try {
            ctx.getCatalogManager()
                    .alterCatalog(
                            catalogName, conf -> conf.addAll(Configuration.fromMap(properties)));

            return TableResultImpl.TABLE_RESULT_OK;
        } catch (CatalogException e) {
            throw new ValidationException(
                    String.format("Could not execute %s", asSummaryString()), e);
        }
    }
```

For `ALTER CATALOG RESET`
```java
    @Override
    public TableResultInternal execute(Context ctx) {
        try {
            ctx.getCatalogManager()
                    .alterCatalog(catalogName, conf -> resetKeys.forEach(conf::removeKey));

            return TableResultImpl.TABLE_RESULT_OK;
        } catch (CatalogException e) {
            throw new ValidationException(
                    String.format("Could not execute %s", asSummaryString()), e);
        }
    }
```

Reply:
Execellent idea~